lib/synchronizer.rb

```diff
@@ -2,25 +2,29 @@ class Synchronizer
 
   attr_reader :backup
 
-	def initialize(backup)
-    @backup             = backup
-    @api_key            = SiteSetting.discourse_backups_api_key
-    @turned_on          = SiteSetting.discourse_backups_enabled
-    @number_of_backups  = SiteSetting.discourse_backups_quantity
-	end
+  def backup
+    @backup
+  end
+
+  def initialize(backup)
+    @backup = backup
+    @api_key = SiteSetting.discourse_backups_api_key
```

Comment:
I think the issue here is that `can_sync?` should not be implemented in the base class. It should only be implemented by the class that extends it. So you don't need to check site settings in this class, let the classes that extend this (GoogleDriveSynchronizer) do all that, using its own site settings.